sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -466,7 +466,7 @@ case class UpdateAction(
       newChildren: IndexedSeq[Expression]): UpdateAction =
     copy(
       condition = if (condition.isDefined) Some(newChildren.head) else None,
-      assignments = newChildren.tail.asInstanceOf[Seq[Assignment]])
+      assignments = newChildren.takeRight(assignments.length).asInstanceOf[Seq[Assignment]])
```

Comment:
fixed a bug here: we should not always exclude the first element, as `condition` can be None

Reply:
cc @sigmod

Reply:
Thanks for the fix, @cloud-fan!

cc @dbaliafroozeh

Reply:
Thanks for the fix @cloud-fan!